conda-store-server/conda_store_server/build.py

```diff
@@ -38,8 +39,11 @@ def set_build_started(db: Session, build: orm.Build):
     db.commit()
 
 
-def set_build_failed(db: Session, build: orm.Build):
+def set_build_failed(
+    db: Session, build: orm.Build, status_info: typing.Optional[str] = None
+):
     build.status = schema.BuildStatus.FAILED
+    build.status_info = status_info
```

Comment:
What's the maximum size allowed? Do we need to check?

Reply:
We don't need to check here as it'd clutter the code. SQLAlchemy will throw an error on inserts if this is a problem for your backend. And the intent here is that error messages that appear in `status_info` are human-written and are user-facing. No one should be passing random exceptions here because that might leak sensitive data depending on the context. You know how popular websites show you "Something went wrong" without explaining why. Or sometimes they might do "Error 12345: something failed". No real info is shared or it's encoded via a number for the dev team to look up later (but we don't have a policy for that at the moment, so not adding the number encoding thing). Same idea here.